File: pairs-of-songs-with-total-durations-divisible-by-60/pairs-of-songs-with-total-durations-divisible-by-60.go
func numPairsDivisibleBy60(time []int) (result int) {
    count := make([]int, 60)
    for _, t := range time {
        count[t%60]++
    }
    l := 1
    h := 59
    for l < h {
        result += count[l] * count[h]
        l++
        h--
    }
    result += (count[0] * (count[0]-1))/2
    result += (count[30] * (count[30]-1))/2
    return
}

File: pairs-of-songs-with-total-durations-divisible-by-60/pairs-of-songs-with-total-durations-divisible-by-60.java
class Solution {
    public int numPairsDivisibleBy60(int[] time) {
        int[] count = new int[60];
        for (int i : time) {
            count[i%60]++;
        }
        int l = 1;
        int h = 59;
        int result = 0;
        while (l < h) {
            result += (count[l++] * count[h--]);
        }
        result += (count[0] * (count[0]-1))/2;
        result += (count[30] * (count[30]-1))/2;
        return result;
    }
}